app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
```

Comment:
I could make the `initialize` private if that's what you're saying.

The reasoning here is very simple and something I've been trying to do, which is not using `new` at all, and always have a more informative substitute for it. This is an ideia from Corey Haines' "4 rules of simple design" book.

In this example, 

```ruby
Subscriber.new(discussion: discussion)
```

makes you assume that you are going to subscribe to the discussion, you are not sure.

```ruby
Subscriber.for_discussion(discussion)
```

As for this case, you can clearly see that you are creating a subscriber for that discussion (at least that was the idea).

<ramble>
If ruby had a similar syntax for named arguments as ios does (particularly swift), you could write something like:

```ruby
Subscriber.new(subscribed_discussion: discussion)

class Subscriber
  def initialize(subscribed_discussion discussion: nil)
    @discussion = discussion
  end
end
```

Thus, `subscribed_discussion` would be the external name for the argument and `discussion` the internal name.

Still it is arguable which one is best.
</ramble>